vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -0,0 +1,147 @@
+package ca.on.oicr.gsi.vidarr.server;
+
+import static java.util.Map.Entry.comparingByValue;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.prometheus.client.Gauge;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+
+final class PriorityByWorkflow implements ConsumableResource {
+
+  private List acceptedPriorities = Arrays.asList(1, 2, 3, 4);
+
+  private static final class WaitingState {
+
+    private final SortedSet<SimpleEntry<String, Integer>> waiting = new ConcurrentSkipListSet<SimpleEntry<String, Integer>>(comparingByValue()){
+
+      @Override
+      public boolean add(SimpleEntry<String, Integer> simpleEntry) {
+        super.remove(this.getByKey(simpleEntry.getKey()));
+        return super.add(simpleEntry);
+      }
+
+      @Override
+      public boolean remove(Object o) {
+        try {
+          String oEntry = (String) o;
+          for (SimpleEntry entry : this){
+            if (entry.getKey().equals(oEntry)) {super.remove(entry);}
+          }
+        } catch (Exception e){
+          return false;
+        }
+        return true;
+      }
+
+      public SimpleEntry getByKey(String simpleEntryKey){
+        for (SimpleEntry entry : this){
+          if (entry.getKey().equals(simpleEntryKey)) {return(entry);}
+        }
+        return null;
+      }
+    };
+  }
+
+
+  //remove?
+  private static final Gauge currentInPriorityWaitingCount =
+      Gauge.build(
+              "vidarr_in_priority_waiting_per_workflow_current",
+              "The current number of workflows that are on priority waiting to run.")
```

Comment:
Would it be accurate to call this `The current number of workflow runs that are waiting to run.`?

Reply:
That would imply this is the count of all workflow runs in the WAITING state rather than the workflow runs in the WAITING state due to this particular consumable resource, wouldn't it? PriorityByWorkflow can only count the workflow runs waiting for itself

Reply:
Ah, that's true. Thanks @avarsava !

Reply:
Good call, is this giving worthwhile info then? Or should I remove to prevent confusion?